File: Nim/K04.nim
import os
import random
import times
from strutils import parseInt
import IExecutable

type K04* = ref object of IExecutable

method question1*(this: K04) =
    stdout.write "年齢を入力してください。＞"
    stdout.flushFile
    var age = parseInt(readLine(stdin))
    if age < 3 or age >= 70:
        echo "入場料金無料です。"
    else:
        echo "通常料金です。"

method question2*(this: K04) =
    stdout.write "性別を選択してください。（０：男性　１：女性）＞"
    stdout.flushFile
    var gender = parseInt(readLine(stdin)).toFloat
    case gender
    of 0:
        echo "あら、格好良いですね。"
    of 1:
        echo "あら、モデルさんみたいですね。"
    else:
        echo "そんな選択肢はありません。"

method question3*(this: K04) =
    stdout.write "年齢を入力してください。＞"
    stdout.flushFile
    var age = parseInt(readLine(stdin))
    if age < 3 or age >= 70:
        echo "入場料金無料です。"
    elif age >= 3 and age <= 15:
        echo "子供料金で半額です。"
    elif age >= 60 and age < 70:
        echo "シニア割引で一割引きです。"
    else:
        echo "通常料金です。"

method question4*(this: K04) =
    echo "＊＊＊おみくじプログラム＊＊＊"
    echo "おみくじを引きますか　（はい：１　いいえ：０）　＞"
    var choice = parseInt(readLine(stdin))

    if choice >= 1:
        randomize(getTime().toWinTime)
        var oracle = rand(4)

        case oracle
        of 0:
            echo "大吉　とってもいいことがありそう！！"
        of 1:
            echo "中吉　きっといいことあるんじゃないかな"
        of 2:
            echo "小吉　少しぐらいはいいことあるかもね"
        of 3:
            echo "凶　今日はおとなしくておいた方がいいかも"
        of 4:
            echo "大凶　これじゃやばくない？早く家に帰った方がいいかも"
        else:
            return

File: Nodejs/K04.ts
import { IExecutable } from './IExecutable';
import * as readline from 'readline';

export class K04 extends IExecutable {

    Question1(rl?: readline.ReadLine): Promise<string> {
        try {

            let age: number;

            let q1 = () => {
                return new Promise((resolve, reject) => {
                    rl.question("年齢を入力して下さい＞", (value) => {
                        age = Number.parseInt(value);
                        resolve();
                    });
                });
            };

            let showResult = () => {
                return new Promise(async (resolve, reject) => {
                    await q1();
                    if (age < 3 || age >= 70) {
                        console.log("入場料金無料です。");
                    } else {
                        console.log("通常料金です。");
                    }
                    resolve();
                });
            };

            return new Promise(async (resolve, reject) => {
                await showResult();
                resolve();
            });

        } catch (e) {
            console.error(e);
        }
    }

    Question2(rl?: readline.ReadLine): Promise<string> {
        try {

            let gender: number;

            let q1 = () => {
                return new Promise((resolve, reject) => {
                    rl.question("性別を選択してください。（０：男性　１：女性）＞", (value) => {
                        gender = Number.parseInt(value);
                        resolve();
                    });
                });
            };

            let showResult = () => {
                return new Promise(async (resolve, reject) => {
                    await q1();
                    switch (gender) {
                        case 0:
                            console.log("あら、格好良いですね。");
                            break;
                        case 1:
                            console.log("あら、モデルさんみたいですね。");
                            break;
                        default:
                            console.log("そんな選択肢はありません。");
                            break;
                    }
                    resolve();
                });
            };

            return new Promise(async (resolve, reject) => {
                await showResult();
                resolve();
            });

        } catch (e) {
            console.error(e);
        }
    }

    Question3(rl?: readline.ReadLine): Promise<string> {
        try {
            let age: number;

            let q1 = () => {
                return new Promise((resolve, reject) => {
                    rl.question("年齢を入力して下さい＞", (value) => {
                        age = Number.parseInt(value);
                        resolve();
                    });
                });
            };

            let showResult = () => {
                return new Promise(async (resolve, reject) => {
                    await q1();
                    if (age < 3 || age >= 70) {
                        console.log("入場料金無料です。");
                    } else if (age >= 3 && age <= 15) {
                        console.log("子供料金で半額です。");
                    } else if (age >= 60 && age < 70) {
                        console.log("シニア割引で一割引きです。");
                    } else {
                        console.log("通常料金です。");
                    }
                    resolve();
                });
            };

            return new Promise(async (resolve, reject) => {
                await showResult();
                resolve();
            });
        } catch (e) {
            console.error(e);
        }
    }

    Question4(rl?: readline.ReadLine): Promise<string> {
        try {
            console.log("＊＊＊おみくじプログラム＊＊＊");

            let choice: number;
            let q1 = () => {
                return new Promise((resolve, reject) => {
                    rl.question("おみくじを引きますか　（はい：１　いいえ：０）　＞", (value) => {
                        choice = Number.parseInt(value);
                        resolve();
                    });
                });
            };

            let showResult = () => {
                return new Promise(async (resolve, reject) => {
                    await q1();
                    if (choice >= 1) {
                        let oracle = Math.round(Math.random() * 4);
                        switch (oracle) {
                            case 0:
                                console.log("大吉　とってもいいことがありそう！！");
                                break;
                            case 1:
                                console.log("中吉　きっといいことあるんじゃないかな");
                                break;
                            case 2:
                                console.log("小吉　少しぐらいはいいことあるかもね");
                                break;
                            case 3:
                                console.log("凶　今日はおとなしくておいた方がいいかも");
                                break;
                            case 4:
                                console.log("大凶　これじゃやばくない？早く家に帰った方がいいかも");
                                break;
                            default:
                                break;
                        }
                    }

                    resolve();
                });
            };

            return new Promise(async (resolve, reject) => {
                await showResult();
                resolve();
            });
        } catch (e) {
            console.error(e);
        }
    }

}